File: back-end/src/main/java/com/lumberjack/quizappbackend/model/quiz/dto/QuizDto.java
package com.lumberjack.quizappbackend.model.quiz.dto;

import com.lumberjack.quizappbackend.model.quiz.TimeLimit;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import java.io.Serial;
import java.io.Serializable;
@Getter
@Setter
@ToString
@NoArgsConstructor
public class QuizDto implements Serializable {
    @Serial
    private static final long serialVersionUID = -4354354354543543554L;

    private String id;
    private String title;
    private String category;
    private TimeLimit timeLimit;
    private String createdAt;
}

File: front-end/src/app/model/quiz-dto.ts
import { Question } from "./question";
import { Quiz } from "./quiz";
import { TimeLimit } from "./time-limit";

export class QuizDto {

  id: string;
  title: string;
  category: string;
  timeLimit: TimeLimit;
  createdAt: string;

  static quizToQuizDto(quiz: Quiz) {
    let quizDto = new QuizDto();
    quizDto.category = quiz.category;
    quizDto.createdAt = quiz.createdAt;
    quizDto.id = quiz.id;
    quizDto.title = quiz.title;
    quizDto.timeLimit = quiz.timeLimit;
    return quizDto;
  }
}
